ProcessLib/Output.cpp

```diff
@@ -134,46 +134,39 @@ void Output::addProcess(ProcessLib::Process const& process,
                                  std::forward_as_tuple(filename));
 }
 
-Output::SingleProcessData Output::findSingleProcessData(
-    Process const& process, const unsigned process_id) const
-{
-    if (process.isMonolithicSchemeUsed())
-    {
-        auto spd_it = _single_process_data.find(&process);
-        if (spd_it == _single_process_data.end()) {
-            OGS_FATAL("The given process is not contained in the output"
-                      " configuration. Aborting.");
-        }
-        return spd_it->second;
-    }
-
-    auto spd_range = _single_process_data.equal_range(&process);
-    unsigned counter = 0;
-    for (auto spd_it=spd_range.first; spd_it!=spd_range.second; ++spd_it)
-    {
-        if(counter == process_id)
-        {
-            return spd_it->second;
-        }
-        counter++;
-    }
-
-    OGS_FATAL("The given process is not contained in the output"
-                      " configuration. Aborting.");
-}
-
-
 void Output::doOutputAlways(Process const& process,
                             const int process_id,
                             ProcessOutput const& process_output,
                             unsigned timestep,
                             const double t,
                             GlobalVector const& x)
 {
+    const bool make_output =
+        !(process_id < static_cast<int>(_single_process_data.size()) - 1 &&
+          !(process.isMonolithicSchemeUsed()));
+    if (!make_output)
+        return;
+ 
     BaseLib::RunTime time_output;
     time_output.start();
 
-    auto spd = findSingleProcessData(process, process_id);
+    auto spd_range = _single_process_data.equal_range(&process);
+    int counter = 0;
+    SingleProcessData* spd_ptr = nullptr;
+    for (auto spd_it=spd_range.first; spd_it!=spd_range.second; ++spd_it)
+    {
+        if(counter == process_id)
```

Comment:
Just curious: Is there a one-to-one correspondence between a `process_id` and a `process` in `_single_process_data`?

Reply:
Yes, they are. However In the present code, each process outputs VTU file with all variables of all processes.  We may need an option that allows one process only output its own variables in future. Therefore, I keep Output::SingleProcessData for this purpose. Otherwise, 
1.  struct  Output::SingleProcessData  can be removed by enabling MeshLib::IO::PVDFile pvd_file as a member of class Output directly.
2. ` std::multimap<Process const*, SingleProcessData> _single_process_data;` can be dropped too.

Reply:
> However In the present code, each process outputs VTU file with all variables of all processes.

But since only the last process writes output, everything is written only once, right?

> We may need an option that allows one process only output its own variables in future.

Not sure, because then the grid would be written even more often than it's written now.

Reply:
>> However In the present code, each process outputs VTU file with all variables of all processes.

> But since only the last process writes output, everything is written only once, right?

Yes, it is.


> >We may need an option that allows one process only output its own variables in future.

>Not sure, because then the grid would be written even more often than it's written now.

If it is not needed, I can remove ``struct Output::SingleProcessData`` in another PR.

Reply:
Yes, a later PR is fine. Maybe, once everything settled, a cleanup of the output logic would be appropriate anyway.

Reply:
Good. Another PR to clean up  the output logic including to remove ``struct Output::SingleProcessData``.